pandas/core/arrays/categorical.py

```diff
@@ -379,6 +382,109 @@ def __init__(
         self._dtype = self._dtype.update_dtype(dtype)
         self._codes = coerce_indexer_dtype(codes, dtype.categories)
 
+    @classmethod
+    def from_dummies(
+        cls, dummies: "DataFrame", ordered: Optional[bool] = None
+    ) -> "Categorical":
+        """Create a `Categorical` using a ``DataFrame`` of dummy variables.
+
+        The ``DataFrame`` must be coercible to boolean,
+        and have no more than one truthy value per row.
+        The columns of the ``DataFrame`` become the categories of the `Categorical`.
+        A column whose header is NA will be dropped;
+        any row with a NA value will be uncategorised.
+
+        Parameters
+        ----------
+        dummies : DataFrame of bool-like
+        ordered : bool
+            Whether or not this Categorical is ordered.
+
+        Raises
+        ------
+        ValueError
+            If a sample belongs to >1 category
+
+        Returns
+        -------
+        Categorical
+
+        Examples
+        --------
+        >>> df = pd.DataFrame(
+        ... [[1, 0, 0], [0, 1, 0], [0, 0, 1]], columns=["a", "b", "c"]
+        ... )
+        >>> Categorical.from_dummies(df)
+        [a, b, c]
+        Categories (3, object): [a, b, c]
+        """
+        df = dummies.drop(columns=np.nan, errors="ignore").astype(bool)
+
+        if (df.sum(axis=1) > 1).any():
+            raise ValueError("Some rows belong to >1 category")
+
+        mult_by = np.arange(df.shape[1]) + 1
+        #  000            000    0   -1
+        #  010            020    2    1
+        #  001 * 1,2,3 => 003 -> 3 -> 2 = correct codes
+        #  100            100    1    0
+        codes = (df * mult_by).sum(axis=1) - 1
+        codes[codes.isna()] = -1
+        return cls.from_codes(codes, df.columns.values, ordered=ordered)
+
+    def to_dummies(self, na_column=None) -> "DataFrame":
```

Comment:
I think this should be called `get_dummies` I don't think the justification for deviating from that name is strong enough to warrant a different name.

I also think it should exactly match the signature of `get_dummies`.